code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -466,14 +480,12 @@
 		turn_cooling_off(H, 1)
 
 /obj/item/weapon/rig/process()
-	// If we've lost any parts, grab them back.
-	var/mob/living/M
-	for(var/obj/item/piece in list(gloves,boots,helmet,chest))
-		if(piece.loc != src && !(wearer && piece.loc == wearer))
-			if(istype(piece.loc, /mob/living))
-				M = piece.loc
-				M.unEquip(piece)
-			piece.forceMove(src)
+	// Not on a mob...?
+	if(!ismob(loc))
+		wearer?.wearing_rig = null

```

Comment:
For maximum oomph, should check that wearer.wearing_rig == src before setting it null.

Reply:
Done